### ISSUE TITLE
storage.hbase.short-cf-names is not fixed

### ISSUE THREAD

**Issue by pluradj** [JanusGraph/janusgraph#220]
Following up on this [discussion thread](https://groups.google.com/d/msg/janusgraph-users/crPTyWRWzZU/fyMTgLnxCAAJ), the HBase [configuration docs](http://docs.janusgraph.org/0.1.0-SNAPSHOT/config-ref.html#_storage_hbase) state that `storage.hbase.short-cf-names` is `FIXED`, meaning that the configuration option cannot be changed for the entire life of the database. The observed behavior sounds more like `MASKABLE` is being applied, which is the incorrect behavior.

Steps to recreate in console:
```
graph = JanusGraphFactory.build().set("storage.backend", "hbase").
        set("storage.hbase.short-cf-names", false).open()
graph.io(graphml()).readGraph("data/tinkerpop-modern.xml")
g = graph.traversal()
// expecting 6 vertices
g.V().valueMap(true)
graph.close()

graph = JanusGraphFactory.build().set("storage.backend", "hbase").
        set("storage.hbase.short-cf-names", true).open()
g = graph.traversal()
// expecting 6 vertices
g.V().valueMap(true)
graph.close()
```

**Comment by jerryjch** [comment]
This does not seem to be an issue. 
'storage.hbase.short-cf-names' is FIXED and can not be changed once the graph is there.  The default is 'true'.
In the mentioned discussion thread, the user set it as non-default 'false'. Then the user needs to set it to 'false' when using the graph everywhere. Otherwise the default 'true' kicks in, which causes not being able to access the graph.

**Comment by pluradj** [comment]
I believe the original configuration is stored within the table. I'd expect a warning issued if the user attempted to connect with an override on a fixed property, then the graph would still open using the original configuration value.

**Comment by pluradj** [comment]
We need to dig into the code some more, but if the behavior really is and should be that the user must override the fixed property every time, the configuration reference documentation for `FIXED` can be made more clear on the behavior and expected usage.

**Comment by jerryjch** [comment]
Looking into it more. Here is what happens.

The short-cf-names flag is checked when HBaseStoreManager tries to open a store. For example, system_properties or edgestore.  And it determines which column family to go to.  For example, either 'system_properties' or 's'.  It the user has used short-cf-names=false.  The column family 'system_properties' has been created in the table and populated. The same for the edgestore.
If next time the user uses  short-cf-names=true, HBaseStoreManager sees the column family 's' does not exist, then it will create the empty column family 's'.  After that, there are two set of column families with one set having the long name and the other set having the short name.  And possible one set has data and the other is empty.

I don't think this is the intention of the short-cf-names flag.  We should probably throw exception if the table already has one set of column family (e.g shortname), then the user tries to open the graph on the same table with long column family names. 
Note that we can not depend on fetching the short-cf-names property from the system properties store because the flag is checked before we know which column family to go to.

**Comment by jerryjch** [comment]
Fixed by #237
